app/models/subsidiary.rb

```diff
@@ -20,7 +20,16 @@ def self.find(id)
   end
 
   def plans
-    Plan.all
+    response = Faraday.get("#{id}/api/plans")
+    if response.status == 200
+      list = JSON.parse(response.body, symbolize_names: true)
+      list.map do |item|
+        Plan.new(id: item[:id], name: item[:name], monthly_payment: item[:monthly_payment],
+                 permanency: item[:permanency], subsidiary: id)
```

Comment:
O item não já é um hash de chave e valor que você recebe tudo? Talvez dê pra simplificar fazendo algo assim:

```suggestion
        Plan.new(item)
```

Reply:
Não todos, o _item_ não contem a chave _subsidiary_ :(
Eu tentei com Plan.new(item, subsidiary: id) mas não funcionou. Mas deve ter um jeito de dar uma simplificada mesmo.
Talvez adicionar :subsidiary => id em todos os "itens" antes da linha do "Plan.new" ?

Reply:
Na verdade não deveríamos mexer no item, imagino eu. Acho que tem que usar aquele splat?
Plan.new({ **item, subsidiary: id }) ?

Reply:
Ele está dando um warning de que usar a hash como key word arguments não é uma boa ideia (deprecated), então teríamos que passar **item para o Plan.new
Ou seja, passei **{ **item, subsidiary: id } :P